Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -451,6 +452,66 @@ bool Interpreter::isOpcode(Opcode opcode)
 }
 #endif // ASSERT_ENABLED
 
+
+void Interpreter::getAsyncStackTrace(JSCell* owner, Vector<StackFrame>& results, JSGenerator* generator, size_t maxStackSize)
+{
+    RELEASE_ASSERT(Options::useAsyncStackTrace());
+    ASSERT(generator);
+
+    AssertNoGC assertNoGC;
+
+    VM& vm = this->vm();
+
+    auto getParentGenerator = [&](JSGenerator* gen) -> JSGenerator* {
+        JSValue contextValue = gen->internalField(static_cast<unsigned>(JSGenerator::Field::Context)).get();
+        JSPromise* awaitedPromise = jsDynamicCast<JSPromise*>(contextValue);
+        if (awaitedPromise && awaitedPromise->status(vm) == JSPromise::Status::Pending) {
+            JSValue reactionsValue = awaitedPromise->internalField(JSPromise::Field::ReactionsOrResult).get();
+            if (JSObject* reactions = jsDynamicCast<JSObject*>(reactionsValue)) {
+                // The @context field is always at offset 3 in Promise reactions objects
+                // created by pushNewPromiseReaction in PromiseOperations.js
+                constexpr PropertyOffset contextPropertyOffset = 3;
+                JSValue contextFieldValue = reactions->getDirect(contextPropertyOffset);
```

Comment:
I think this is also not guaranteed because how object gets layout can be even changed via object allocation profile. I think we need a specific object with internal fields instead of using the normal object for `ReactionsOrResult`.